experiment_analysis/plotting.py

```diff
@@ -95,16 +98,34 @@ def get_timeseries(
                                                                     metric=f"{metric}_cuped",
                                                                     method=CappingMethod.WITH_CUPED,
                                                                     metric_raw=metric)
+        elif method == TimeseriesMethod.ISOTONIC:
+            params.update({"metric": f"{metric}_isotonic"})
+            df_agg_joined = df_agg_metric.merge(
+                df_pre_exp[[*merge_cols, f"{covariate_prefix}_{metric}"]], on=merge_cols, how="left")
+            df_agg_joined[f"pre_exp_{metric}"].fillna(0, inplace=True)
+            df_agg_joined[f"{metric}_isotonic"] = apply_isotonic(
+                df_agg_joined, metric, f"{covariate_prefix}_{metric}")
+        elif method == TimeseriesMethod.ISOTONIC_CAPPED:
+            params.update({"metric": f"{metric}_isotonic_capped"})
+            df_agg_joined = df_agg_metric.merge(
+                df_pre_exp[[*merge_cols, f"{covariate_prefix}_{metric}"]], on=merge_cols, how="left")
+            df_agg_joined[f"pre_exp_{metric}"].fillna(0, inplace=True)
+            df_agg_joined[f"{metric}_isotonic"] = apply_isotonic(
+                df_agg_joined, metric, f"{covariate_prefix}_{metric}")
+            df_agg_joined[f"{metric}_isotonic_capped"] = apply_capping(df=df_agg_joined,
+                                                                    metric=f"{metric}_isotoic",
+                                                                    method=CappingMethod.WITH_ISOTONIC,
+                                                                    metric_raw=metric)
 
         # Calculate ttest results for desired metric
-        if method == TimeseriesMethod.CUPED or method == TimeseriesMethod.CUPED_CAPPED:
-            row = summarize_ttest(df_agg_joined, **params)
-            row["ds"] = _date - np.timedelta64(1, 'D')
-            row["bucket_count"] = df_agg_joined.shape[0]
-        else:
+        if method == TimeseriesMethod.CAPPED:
```

Comment:
did you mean `TimeseriesMethod.STANDARD`? I'm assuming that for all the others, we'd want to use the joined data.